portfolio/src/main/webapp/script.js

```diff
@@ -43,3 +43,28 @@ function addCoursework(){
     const courseContainer = document.getElementById('course-container');
     courseContainer.appendChild(courselist);
 }
+async function helloJamila(){
+    const response = await fetch('/data');
+    const greeting = await response.text();
+    document.getElementById('greeting').innerText = greeting;
+}
+async function fetchJSON(){
+    const container = document.getElementById('greeting');
+    console.log("Fetching JSON...");
+    const data = fetch('/data')
+    .then(response => response.json())
+    .then(message => {
+        console.log(message[0]);
+        console.log(message[1]);
+        console.log(message[2]);
+        container.innerText = message[0] + "\n" + message[1] + "\n" + message[2];
+        }
+    );
+}
+// fetch('/my-data-url')  // sends a request to /my-data-url
```

Comment:
You can remove this commented out section if you're comfortable with the json steps now